app/services/media-gallery/media-gallery.ts

```diff
@@ -46,62 +49,63 @@ const concatUint8Arrays = (a: Uint8Array, b: Uint8Array) => {
   return c;
 };
 
-export class MediaGalleryService extends StatefulService<IMediaGalleryState> {
-  @Inject() userService: UserService;
-  @Inject() hostsService: HostsService;
-  @Inject() windowsService: WindowsService;
+export class MediaGalleryService extends Service {
+  @Inject() private userService: UserService;
+  @Inject() private hostsService: HostsService;
+  @Inject() private windowsService: WindowsService;
 
-  static initialState: IMediaGalleryState = {
-    uploads: [],
+  static initialState: IMediaGalleryInfo = {
+    files: [],
     totalUsage: 0,
     maxUsage: null,
     maxFileSize: null
   };
 
   private promises: Dictionary<{
-    resolve: (value?: IFile | PromiseLike<IFile>) => void;
+    resolve: (value?: IMediaGalleryFile) => void;
     reject: () => void;
   }> = {};
 
-  init() {
-    this.fetchFileLimits();
-    this.fetchFiles();
+  private stockImages = stockImages.map(item => {
+    return {
+      ...item,
+      type: 'image',
+      isStock: true,
+      size: 0
+    };
+  });
+
+  private stockSounds = stockSounds.map(item => {
+    return {
+      ...item,
+      type: 'audio',
+      isStock: true,
+      size: 0
+    };
+  });
+
+  async fetchGalleryInfo(): Promise<IMediaGalleryInfo> {
+    const [files, limits] = await Promise.all([this.fetchFiles(), this.fetchFileLimits()]);
+    const totalUsage = files.reduce((size: number, file: IMediaGalleryFile) => size + file.size, 0);
+    return { files, totalUsage, ...limits };
   }
 
-  files() {
-    let totalUsage = 0;
-    const files = this.state.uploads.map((item: IFile) => {
-      const filename = decodeURIComponent(item.href.split(/[\\/]/).pop());
-      const ext = filename
-        .toLowerCase()
-        .split('.')
-        .pop();
-      if (item.size) {
-        totalUsage += parseInt(item.size, 10);
-      }
-
-      item.filename = filename;
-      item.type = filetypeMap[ext];
-      return item;
+  async pickFile(): Promise<IMediaGalleryFile> {
+    const promiseId = uuid();
```

Comment:
is there a reason to use `uuid()` here instead of `ipcRenderer.sendSync('getUniqueId');`?

Reply:
ipcRenderer.sendSync('getUniqueId'); is a blocking IPC request, that is not too fast